Sources/Networking/Misc/DownloadState.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  DownloadState.swift
+//  
+//
+//  Created by Matej Molnár on 07.03.2023.
+//
+
+import Foundation
+
+public extension URLSessionTask {
+    struct DownloadState {
+        public var totalBytesWritten: Int64
```

Comment:
I would vote for better naming here, something like totalBytes vs downloadedBytes?